pkg/cmd/auth/login/login.go

```diff
@@ -283,26 +290,92 @@ func loginRun(opts *LoginOptions) error {
 		fmt.Fprintf(opts.IO.ErrOut, "%s Configured git protocol\n", cs.SuccessIcon())
 	}
 
-	apiClient, err := client.ClientFromCfg(hostname, cfg)
-	if err != nil {
-		return err
-	}
+	var username string
+	if userValidated {
+		username, _ = cfg.Get(hostname, "user")
+	} else {
+		apiClient, err := client.ClientFromCfg(hostname, cfg)
+		if err != nil {
+			return err
+		}
 
-	username, err := api.CurrentLoginName(apiClient, hostname)
-	if err != nil {
-		return fmt.Errorf("error using api: %w", err)
-	}
+		username, err = api.CurrentLoginName(apiClient, hostname)
+		if err != nil {
+			return fmt.Errorf("error using api: %w", err)
+		}
 
-	err = cfg.Set(hostname, "user", username)
-	if err != nil {
-		return err
+		err = cfg.Set(hostname, "user", username)
+		if err != nil {
+			return err
+		}
 	}
 
 	err = cfg.Write()
 	if err != nil {
 		return err
 	}
 
+	if opts.Interactive && gitProtocol == "https" {
+		helper, _ := gitCredentialHelper(hostname)
+		if !isOurCredentialHelper(helper) {
+			var primeCredentials bool
+			err = prompt.SurveyAskOne(&survey.Confirm{
+				Message: "Set up git for passwordless push/pull operations?",
```

Comment:
Was thinking something similar to @ampinsk - curious how something like this feels.
```suggestion
				Message: "Also set up Git authentication for GitHub CLI operations that leverage Git?",
```

Reply:
bitbox rocker

Reply:
@billygriffin The idea is that more git operations than just ones initiated by `gh` are affected. For example, after doing this step, you can safely `git clone <url>` (bypassing GitHub CLI) by using the credentials that you've established during `gh auth login`.

We could also ensure that only `gh`-initiated git operations (e.g. `git clone` within `gh repo clone`, `git push` within `gh pr create`), and that regular `git` operations that you initiate outside of `gh` are unaffected (and possibly un-authenticated). This approach would be simpler to achieve, but I think it would be worse for the user experience, because GitHub CLI often relies on regular `git` operations to be combined with it during everyday workflows.

Something that I've noticed in both hub and GitHub CLI: after users log into GitHub from the command line, they expect to be logged into GitHub for _all_ their GitHub-related operations. To meet their expectations, I think that in this case it's worth leaving the boundaries of our own tool and making sure things work for them even when they were not explicitly using our tool.

Reply:
How about `Authenticate Git with your GitHub credentials?`